test/e2e/framework/volume/fixtures.go

```diff
@@ -663,7 +663,9 @@ func InjectContent(f *framework.Framework, config TestConfig, fsGroup *int64, fs
 // generateWriteCmd is used by generateWriteBlockCmd and generateWriteFileCmd
 func generateWriteCmd(content, path string) []string {
 	var commands []string
-	commands = []string{"/bin/sh", "-c", "echo '" + content + "' > " + path}
+	// flush data so volume can be immediately used as a data source
+	commands = []string{"/bin/sh", "-c", "echo '" + content + "' > " + path +
```

Comment:
Should we also remove the wait for VolumeAttachment?

Reply:
Done

Reply:
Sorry I was rereading  the original issue and I think the "wait for detach" logic was added because of rate limiting issues. So I think we should keep the logic here and maybe mark those tests as "offline" cloning tests    

I believe the multivolume tests you referenced in the beginning is actually the "online" case right? Maybe those can satisfy https://github.com/kubernetes/kubernetes/issues/101333

Reply:
Marking the tests as "offline" cloning would be just adding the offline word, right?  As in changing

```
	ginkgo.It("should provision storage with pvc data source", func() {
```
with
```
	ginkgo.It("should provision storage with offline pvc data source", func() {
```

The multivolume tests I mention are not the online case.  For example, the volume case writes the volume contents using the same `preparePVCDataSourceForProvisioning` function as these 2 tests, and that function waits for the injector pod to disappear, which means that the cloning can happen while the volume is still being detached.  And it's only after the PVC clone is created that multiple pods are created on the same node to attach the PVCs.  So those tests would also need to wait for the volume as these tests if what PR https://github.com/kubernetes/kubernetes/pull/106322 says is right: 

> The non parallel test fails because for the GCE PD Driver, cloning fails if the source disk has another operation going on, so this PR adds an extra check that the disk is ready before we try to clone.

Reply:
sorry I lost track of this. Yes I mean putting "offline" in the description. And I guess since these multivolume tests are also offline, we would want the same approach.